examples/jsm/nodes/utils/ArrayElementNode.js

```diff
@@ -19,18 +19,46 @@ class ArrayElementNode extends Node { // @TODO: If extending from TempNode it br
 
 	}
 
-	generate( builder ) {
+	setup( builder ) {
+
+		if ( this.node.isStorageBufferNode && ! builder.isAvailable( 'storageBuffer' ) ) {
+
+			if ( ! this.node.instanceIndex ) {
+
+				builder.setupPBONode( this.node );
+
+			}
+
+
+		}
+
+		super.setup( builder );
+
+
+	}
+
+	generate( builder, output ) {
 
 		const nodeSnippet = this.node.build( builder );
 		const indexSnippet = this.indexNode.build( builder, 'uint' );
 
+		let snippet = `${nodeSnippet}[ ${indexSnippet} ]`;
 		if ( this.node.isStorageBufferNode && ! builder.isAvailable( 'storageBuffer' ) ) {
 
-			return nodeSnippet;
+			snippet = nodeSnippet;
+
+			// TODO: How to properly detect if the node will be used as an assign target?
```

Comment:
I think you should just check if `output` is `'void'` (i.e. just change `'assign'` to `'void'` here and in AssignNode).